ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -162,13 +160,11 @@ Tensor to_layout_impl(
             SmallVector<uint32_t> padded_output_shape;
 
             for (int index = 0; index < tensor.get_shape().rank(); ++index) {
-                if (index == tensor.get_shape().rank() - 2) { // h dim
-                    padded_output_shape.push_back(ttnn::pad_to_multiple_of_tile_size(tensor.get_shape()[index], tile.get_tile_shape()[0]));
-                } else if (index == tensor.get_shape().rank() - 1) { // w dim
-                    padded_output_shape.push_back(ttnn::pad_to_multiple_of_tile_size(tensor.get_shape()[index], tile.get_tile_shape()[1]));
-                } else {
-                    padded_output_shape.push_back(tensor.get_shape()[index]);
-                }
+                padded_output_shape.push_back(
+                    (index == tensor.get_shape().rank() - 2) ? ttnn::pad_to_multiple_of_tile_size(tensor.get_shape()[index], tile.get_tile_shape()[0]) :
+                    (index == tensor.get_shape().rank() - 1) ? ttnn::pad_to_multiple_of_tile_size(tensor.get_shape()[index], tile.get_tile_shape()[1]) :
+                    tensor.get_shape()[index]
```

Comment:
Better.

Consider doing something like
```
    uint32_t second_last_rank = tensor.get_shape().rank() - 2; // h dim
    uint32_t padded_value = index < second_last_rank ? tensor.get_shape()[index] : ttnn::pad_to_multiple_of_tile_size(tensor.get_shape()[index], index == second_last_rank ? tile.get_tile_shape()[0] : tile.get_tile_shape()[1]);
    padded_output_shape.push_back(padded_value);
```
or
```
    uint32_t second_last_rank = tensor.get_shape().rank() - 2; // h dim
    uint32_t padded_value;
    if (index < second_last_rank) {
        padded_value = tensor.get_shape()[index];
    } else {
        // choose between h dim and w dim
        padded_value = ttnn::pad_to_multiple_of_tile_size(tensor.get_shape()[index], index == second_last_rank ? tile.get_tile_shape()[0] : tile.get_tile_shape()[1]);
    }
    padded_output_shape.push_back(padded_value);
```
here and below.

Also, I'm not sure if the two loops should be a separate function. Seems like the function would be too small until it is needed in a third place.